src/consensus/aft/raft.h

```diff
@@ -962,14 +960,13 @@ namespace aft
         end_idx,
         state->commit_idx);
 
-      AppendEntries ae = {
-        {raft_append_entries},
-        {end_idx, prev_idx},
-        state->current_view,
-        prev_term,
-        state->commit_idx,
-        term_of_idx,
-        contains_new_view};
+      AppendEntries ae;
+      ae.idx = end_idx;
+      ae.prev_idx = prev_idx;
+      ae.term = state->current_view;
+      ae.prev_term = prev_term;
+      ae.leader_commit_idx = state->commit_idx;
+      ae.term_of_idx = term_of_idx;
```

Comment:
```suggestion
      AppendEntries ae{
        .idx = end_idx,
        .prev_idx = prev_idx,
        .term = state->current_view,
        .prev_term = prev_term,
        .leader_commit_idx = state->commit_idx,
        .term_of_idx = term_of_idx
      }
```

seems to work even with clang-11

Reply:
Sure, I'll switch to designated initializers. This instance becomes a little more awkward because of the constraints of designated initializers wrt base members. As written, it's a build error:

```
../src/consensus/aft/raft.h:964:10: error: field designator 'idx' does not refer to any field in type 'aft::AppendEntries'
        .idx = end_idx,
```

So you have to mix designated and non-designated, and squash the corresponding warning:
```
 warning: mixture of designated and non-designated initializers in the same initializer list is a C99 extension [-Wc99-designator]
```
```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wc99-designator"
      AppendEntries ae{
        {},
        {.idx = end_idx, .prev_idx = prev_idx},
        .term = state->current_view,
        .prev_term = prev_term,
        .leader_commit_idx = state->commit_idx,
        .term_of_idx = term_of_idx,
      };
#pragma clang diagnostic pop
```

I think the multiple base classes here are unnecessary, but I'll look at unpicking that separately.

Reply:
Ah that's unfortunate, you are right. I do agree that the hierarchy can be simplified here, but as you say best done separately, and perhaps after the functional changes that are on the way.